crypto/evp/pmeth_gn.c

```diff
@@ -37,11 +37,12 @@ static int gen_init(EVP_PKEY_CTX *ctx, int operation)
     case EVP_PKEY_OP_PARAMGEN:
         ctx->op.keymgmt.genctx =
             evp_keymgmt_gen_init(ctx->keymgmt,
-                                 OSSL_KEYMGMT_SELECT_ALL_PARAMETERS);
+                                 OSSL_KEYMGMT_SELECT_ALL_PARAMETERS, NULL);
```

Comment:
We're really avoiding the obvious `EVP_PKEY_paramgen_init_ex` and `EVP_PKEY_keygen_init_ex`, aren't we? :wink:

Reply:
Do we want all of these functions created?  I'm wondering which are useful and which aren't.

Reply:
I'm neutral to that question, really.

Reply:
Adding e.g. EVP_PKEY_decrypt_init_ex(), lets the application pass params across as part of the init call.  There are helper functions for these which are far easier to use.  Why would a user want to use the param form?

Okay, the helpers use ctrl calls which convert to params and for a new unknown algorithm there are no helpers but neither seems compelling.

Reply:
> Why would a user want to use the param form?

Er, I'm lost, isn't the answer that question the exact reason for modifying `EVP_MAC_init()` (#14310)?  Why would the reason be any different for param / key generation?

Reply:
We don't supply an alternative for MACs.

Reply:
[shrug]
Like I said, I'm neutral...  I thought I remembered the talking on the OTC meeting leaning toward "across the board" rather than just MACs and KDFs, but I may misremember things.  I leave it to you to choose...  and really, should someone feel anything different than what you've done here, it can be added later.

Reply:
I'm intending to add the params argument to the libcrypto/provider API.  What I'm not so sure about is adding the various _ex calls that use it immediately.